frontend/src/views/Storage/StorageModal.tsx

```diff
@@ -13,55 +13,58 @@ import {
   Select,
   TextField,
 } from '@bcgov/design-system-react-components';
-import {
-  formCss,
-  modalHeaderStyle,
-  modalPaddingStyle,
-  formGridBreakpoints,
-} from '../../common.styles';
+import { formCss, modalHeaderStyle, modalPaddingStyle } from '../../common.styles';
 import manureTypeOptions from '@/constants/ManureTypeOptions';
-import { NMPFileImportedManureData } from '@/types';
+import YesNoRadioButtons from '@/components/common/YesNoRadioButtons/YesNoRadioButtons';
+import useAppState from '@/hooks/useAppState';
+import { booleanChecker } from '@/utils/utils';
+import type { StorageForm } from './Storage';
 
-// NMPFileImportedManureData is not the final type
-// TODO: determine type of data for initialModalData
 type ModalComponentProps = {
-  initialModalData: NMPFileImportedManureData;
+  initialModalData: StorageForm;
   handleDialogClose: () => void;
-  handleSubmit: (formData: NMPFileImportedManureData) => void;
-  storageList: any;
+  handleSubmit: (formData: StorageForm) => void;
 };
 
 export default function StorageModal({
   initialModalData,
   handleDialogClose,
   handleSubmit,
-  storageList,
   ...props
 }: ModalComponentProps & ComponentProps<typeof Modal>) {
-  const [formData, setFormData] = useState<NMPFileImportedManureData>(initialModalData);
-  const [isEditingExistingEntry] = useState<boolean>(!!initialModalData?.UniqueMaterialName);
+  const { state } = useAppState();
+  const [formData, setFormData] = useState<StorageForm>(initialModalData);
+
+  const [manureList, setManureList] = useState<{ id: string; label: string }[]>([]);
+
+  // gets imported and generated manures then filters manures based on manure type
+  useEffect(() => {
+    const imported = state.nmpFile.years[0]?.ImportedManures ?? [];
+    const generated = state.nmpFile.years[0]?.GeneratedManures ?? [];
+    const manures = [...imported, ...generated];
 
-  const notUniqueNameCheck = () => {
-    if (isEditingExistingEntry) return false;
-    return storageList.some(
-      (ele: NMPFileImportedManureData) => ele.UniqueMaterialName === formData.UniqueMaterialName,
+    const filteredManures = manures.filter(
+      (manure) => manure.ManureTypeName === formData?.ManureType,
     );
-  };
+
+    const items = filteredManures.map((manure, index) => ({
+      id: `${manure.UniqueMaterialName}-${index}`,
+      label: manure.ManagedManureName ?? '',
```

Comment:
I think `ManagedManureName` should be a required field if we're using it as a label.